### PR TITLE
Vault inflation attack

### PR DESCRIPTION
**Pull request by DaigaroCota**
This pull request addresses vulnerabilities M-1 of Macro audit report.

### CODE REVIEW

**Review by pedrovalido — COMMENTED** [review]
Friendly reminder about the TODO's

**Comment by pedrovalido** [comment]
Lets have a brainstorm session to try and attack the newly implemented fix. Action to consider:
- deposit
- withdraw
- payback
- borrow
- transfer
- burn
- malicious providers
- etc (open to more suggetions)

@DaigaroCota

**Review by pedrovalido — COMMENTED** [review]
When a certain amount of time passes and the vault gains interest, the vault will steal the user's interest.
Flow: 
Vault has 100 shares in the beginning.
Vault has 100 fake assets in the beginning.
User deposits 100 assets
User get 100 shares
Sometime passes and the user's money gains 10 assets of interest.

User's balance would be 105 instead of 110

**Review by pedrovalido — COMMENTED** [review]
When calling the deposit function with the vault's address as the receiver, those funds will get stuck in the router

**Comment by pedrovalido** [comment]
Possible solution: when deploying a vault, use the initializeVaultShares function to also deposit a certain amount of real assets instead of faking.

File: packages/protocol/src/abstracts/BaseVault.sol [hunk 1]
```diff
@@ -216,6 +238,17 @@ abstract contract BaseVault is ERC20, SystemAccessControl, PausableVault, VaultP
...
+   * @dev Extended to avoid burning of address(this) shares created in
+   * `_initializeVaultShares`. See OZ implementation: {ERC20-_burn}.
+   */
+  function __burn(address account, uint256 amount) internal virtual {
+    if (account == address(this)) {
+      revert BaseVault__burn_cannotBurn();
+    }
+    _burn(account, amount);
+  }
+
```
Comment (pedrovalido) [inline packages/protocol/src/abstracts/BaseVault.sol:251]: remove this. Its not going to be used

**Comment by DaigaroCota** [comment]
This pull request was overriden by #541 .

**Pull request closed without merging**